tests/tensor_ops/test_setitem.py

```diff
@@ -59,6 +63,71 @@ def setitem(x, y, index):
     return x_copy
 
 
+class set_item_test_factory:
+    def __init__(
+        self,
+        array_strat: st.SearchStrategy[np.ndarray],
+        index_strat: Callable[[np.ndarray], st.SearchStrategy],
+        value_strat: Callable[[np.ndarray], st.SearchStrategy[np.ndarray]],
+    ):
+        self.array_strat = array_strat
+        self.index_strat = index_strat
+        self.value_strat = value_strat
+
+    def __call__(self, f):
```

Comment:
I assume it would be too messy to type-hint `f` here

Reply:
Good call. The hint here isn't so bad since these functions don't actually accept or return anything:

```python
def __call__(self, f: Callable) -> Callable[[], None]:
```